--- Makefile.in.orig	2021-01-07 08:05:37.000000000 -0800
+++ Makefile.in	2021-01-07 08:30:05.000000000 -0800
@@ -94,6 +94,14 @@
 build_triplet = @build@
 host_triplet = @host@
 noinst_PROGRAMS = astbad$(EXEEXT)
+
+cross_compiling = @cross_compiling@
+ifeq ($(cross_compiling),yes)
+  src_run = 
+else
+  src_run = ./
+endif
+
 TESTS = ast_test$(EXEEXT)
 check_PROGRAMS = ast_test$(EXEEXT)
 subdir = .
@@ -4476,9 +4484,9 @@
 .PRECIOUS: Makefile
 
 AST_PAR: ast_par.source astbad
-	sed -e 's/<AST__BAD>/'`./astbad AST__BAD | tr 'E' 'D'`'/' \
-	    -e 's/<AST__NAN>/'`./astbad AST__NAN | tr 'E' 'D'`'/' \
-	    -e 's/<AST__NANR>/'`./astbad AST__NANF | tr 'E' 'D'`'/' \
+	sed -e 's/<AST__BAD>/'`$(src_run)astbad AST__BAD | tr 'E' 'D'`'/' \
+	    -e 's/<AST__NAN>/'`$(src_run)astbad AST__NAN | tr 'E' 'D'`'/' \
+	    -e 's/<AST__NANR>/'`$(src_run)astbad AST__NANF | tr 'E' 'D'`'/' \
 	    ast_par.source >$@
 
 # ast.h depends on the error-facility files.  ast.h _could_ be made
