--- Makefile.in.orig	2021-10-29 13:14:11.000000000 +0100
+++ Makefile.in	2021-10-29 13:18:59.000000000 +0100
@@ -93,6 +93,14 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
+
+cross_compiling = @cross_compiling@
+ifeq ($(cross_compiling),yes)
+  src_run =
+else
+  src_run = ./
+endif
+
 TESTS = ast_test$(EXEEXT)
 check_PROGRAMS = ast_test$(EXEEXT)
 subdir = .
@@ -4526,9 +4534,9 @@
 .PRECIOUS: Makefile
 
 @NOFORTRAN_FALSE@AST_PAR: $(srcdir)/ast_par.source astbad$(EXEEXT)
-@NOFORTRAN_FALSE@	sed -e 's/<AST__BAD>/'`./astbad$(EXEEXT) AST__BAD | tr 'E' 'D'`'/' \
-@NOFORTRAN_FALSE@	    -e 's/<AST__NAN>/'`./astbad$(EXEEXT) AST__NAN | tr 'E' 'D'`'/' \
-@NOFORTRAN_FALSE@	    -e 's/<AST__NANR>/'`./astbad$(EXEEXT) AST__NANF | tr 'E' 'D'`'/' \
+@NOFORTRAN_FALSE@	sed -e 's/<AST__BAD>/'`$(src_run)astbad$(EXEEXT) AST__BAD | tr 'E' 'D'`'/' \
+@NOFORTRAN_FALSE@	    -e 's/<AST__NAN>/'`$(src_run)astbad$(EXEEXT) AST__NAN | tr 'E' 'D'`'/' \
+@NOFORTRAN_FALSE@	    -e 's/<AST__NANR>/'`$(src_run)astbad$(EXEEXT) AST__NANF | tr 'E' 'D'`'/' \
 @NOFORTRAN_FALSE@	    $(srcdir)/ast_par.source >$@
 
 # ast.h depends on the error-facility files.  ast.h _could_ be made

