--- Makefile.in.orig	2025-09-04 14:01:27
+++ Makefile.in	2025-09-04 14:00:53
@@ -93,6 +93,14 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
+
+cross_compiling = @cross_compiling@
+ifeq ($(cross_compiling),yes)
+  src_run =
+else
+  src_run = ./
+endif
+
 TESTS = ast_test$(EXEEXT)
 check_PROGRAMS = ast_test$(EXEEXT)
 subdir = .
@@ -440,8 +448,8 @@
 	libast_grf_2.0.la libast_grf3d.la libast_err.la
 am__astbad_SOURCES_DIST = src/astbad.c src/pointset.h
 @NOFORTRAN_FALSE@am_astbad_OBJECTS = src/astbad.$(OBJEXT)
-astbad_OBJECTS = $(am_astbad_OBJECTS)
-astbad_LDADD = $(LDADD)
+@NOFORTRAN_FALSE@astbad_OBJECTS = $(am_astbad_OBJECTS)
+@NOFORTRAN_FALSE@astbad_LDADD = $(LDADD)
 SCRIPTS = $(bin_SCRIPTS) $(dist_bin_SCRIPTS) $(dist_noinst_SCRIPTS) \
 	$(noinst_SCRIPTS)
 AM_V_P = $(am__v_P_@AM_V@)
@@ -2201,9 +2209,9 @@
 src/astbad.$(OBJEXT): src/$(am__dirstamp) \
 	src/$(DEPDIR)/$(am__dirstamp)
 
-astbad$(EXEEXT): $(astbad_OBJECTS) $(astbad_DEPENDENCIES) $(EXTRA_astbad_DEPENDENCIES) 
-	@rm -f astbad$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(astbad_LDFLAGS) $(astbad_OBJECTS) $(astbad_LDADD) $(LIBS) 
+@NOFORTRAN_FALSE@astbad$(EXEEXT): $(astbad_OBJECTS) $(astbad_DEPENDENCIES) $(EXTRA_astbad_DEPENDENCIES)
+@NOFORTRAN_FALSE@	@rm -f astbad$(EXEEXT)
+@NOFORTRAN_FALSE@	$(AM_V_CCLD)$(LINK) $(astbad_LDFLAGS) $(astbad_OBJECTS) $(astbad_LDADD) $(LIBS)
 install-binSCRIPTS: $(bin_SCRIPTS)
 	@$(NORMAL_INSTALL)
 	@list='$(bin_SCRIPTS)'; test -n "$(bindir)" || list=; \
@@ -4524,9 +4532,9 @@
 .PRECIOUS: Makefile
 
 @NOFORTRAN_FALSE@AST_PAR: $(srcdir)/ast_par.source astbad$(EXEEXT)
-@NOFORTRAN_FALSE@	ASTBAD=`./astbad$(EXEEXT) AST__BAD` && \
-@NOFORTRAN_FALSE@	ASTNAN=`./astbad$(EXEEXT) AST__NAN` && \
-@NOFORTRAN_FALSE@	ASTNANF=`./astbad$(EXEEXT) AST__NANF` && \
+@NOFORTRAN_FALSE@	ASTBAD=`$(src_run)/astbad$(EXEEXT) AST__BAD` && \
+@NOFORTRAN_FALSE@	ASTNAN=`$(src_run)/astbad$(EXEEXT) AST__NAN` && \
+@NOFORTRAN_FALSE@	ASTNANF=`$(src_run)/astbad$(EXEEXT) AST__NANF` && \
 @NOFORTRAN_FALSE@	sed -e 's/<AST__BAD>/'`echo $$ASTBAD | tr 'E' 'D'`'/' \
 @NOFORTRAN_FALSE@	    -e 's/<AST__NAN>/'`echo $$ASTNAN | tr 'E' 'D'`'/' \
 @NOFORTRAN_FALSE@	    -e 's/<AST__NANR>/'`echo $$ASTNANF | tr 'E' 'D'`'/' \
