--- Makefile.in.orig	2021-11-08 20:11:53.098084535 -0800
+++ Makefile.in	2021-11-08 20:20:48.559434015 -0800
@@ -1572,7 +1572,7 @@
     $(WCSLIB_FILES) \
     ast_err.h
 
-libast_la_LDFLAGS = -version-info @version_info@
+libast_la_LDFLAGS = -version-info @version_info@ -lm
 @EXTERNAL_CMINPACK_FALSE@@EXTERNAL_PAL_FALSE@libast_la_LIBADD = libast_pal.la libast_cminpack.la
 @EXTERNAL_CMINPACK_FALSE@@EXTERNAL_PAL_TRUE@libast_la_LIBADD = $(libdir)/libpal.la libast_cminpack.la
 @EXTERNAL_CMINPACK_TRUE@@EXTERNAL_PAL_FALSE@libast_la_LIBADD = libast_pal.la -lcminpack
@@ -1633,7 +1633,7 @@
 
 #  Positional astronomy libraries.
 libast_pal_la_SOURCES = $(PAL_FILES)
-libast_pal_la_LDFLAGS = -version-info @version_info@
+libast_pal_la_LDFLAGS = -version-info @version_info@ -lm
 
 #  Cminpack
 libast_cminpack_la_SOURCES = $(CMINPACK_FILES)
